### PR TITLE
YDA-5789: limit max number of client connections

### PR DESCRIPTION
**Pull request by lwesterhof**
https://github.com/UtrechtUniversity/yoda/pull/445

What is an acceptable default max user connections?

### CODE REVIEW

File: util/config.py [hunk 1]
```diff
@@ -145,6 +145,7 @@ def __repr__(self):
                 vault_copy_backoff_time=300,
                 vault_copy_max_retries=5,
                 vault_copy_multithread_enabled=True,
+                user_max_connections=3,
```
Comment (claravox) [inline util/config.py]:
```suggestion
                user_max_connections=4,
```

**Review by claravox — APPROVED** [review]
LGTM

File: util/user.py [hunk 2]
```diff
@@ -104,3 +105,16 @@ def name_from_id(ctx, user_id):
...
+
+def number_of_connections(ctx):
+    """Get number of active connections from client user."""
+    connections = 0
+    try:
+        ips = subprocess.check_output(["ips", "-a"])
+        username = session_vars.get_map(ctx.rei)['client_user']['user_name']
+        connections = ips.count(username)
+    except Exception:
+        return 0
```
Comment (stsnel) [inline util/user.py:123]: I think it would be good to also log the exception here, since this is an unexpected situation.

**Review by stsnel — COMMENTED** [review]
LGTM :+1:

**Review by stsnel — APPROVED** [review]

**Review by lwesterhof — COMMENTED** [review]
LGTM

**Pull request merged**